core/internal/cltest/cltest.go

```diff
@@ -619,6 +619,10 @@ type User struct {
 func (ta *TestApplication) NewHTTPClient(user *User) HTTPClientCleaner {
 	ta.t.Helper()
 
+	if user == nil {
```

Comment:
In what cases would `user` not be `nil` at this step in the `NewHTTPClient` logic?